### PR TITLE
TEG can power itself when turned on

### PR DESCRIPTION
**Pull request by WarMechanic**
## About the PR

The TEG now redirects 1000w to power itself first after being bootstrapped, and sends the rest to the station.

## Why / Balance

TEG requiring APC power meant that the TEG would brown-out with changes to its LV power grid, and you could place an APC on top of the TEG as a bootstrapper to trivialise this mechanic. Now the TEG powers itself such that you no longer need to isolate the LV grid it is on, however it still cannot cold start and needs APC power to bootstrap it.

## Technical details

This PR modifies `C.S/Power/Generation/Teg/TegSystem.cs` by subtracting the power from its load (clamped to 0+) and subtracting its max load of 1000 from its HV power output (clamped to 0+)

## Media

https://youtu.be/z8o9M4ISlO0
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

None

**Changelog**

:cl:
- tweak: The TEG now powers itself once started.

### CODE REVIEW

**Comment by WarMechanic** [comment]
@PJB3005 i'd like to apologise because one of my arguments earlier was actually incorrect, in that the teg's yellow static light turns off when the teg isn't active which gives you a way to identify whether the teg is on or off.

in any case i think this solution is more elegant than the last one so i'd say we've met a pretty good compromise here, thanks!

**Comment by WarMechanic** [comment]
if the teg output falls below 1kw then the teg will turn off
image

**Comment by Partmedia** [comment]
Thanks for the PR. The code seems okay, though I would prefer that the value '1000' get split out into a const float.

That said I'm not sure that this mechanic is needed or better.

> you could place an APC on top of the TEG as a bootstrapper to trivialise this mechanic

Players who understand what's going on and take initiative to anticipate problems before they occur should be rewarded, no?

And the TEG generates HV, but requires LV. To be consistent with how our universe works, you can't just use HV if you need LV. If this is allowed without a substation, why does any other LV consume require a substation?

**Comment by deltanedas** [comment]
the lv can be for circuitry which obviously wouldnt be using kV of power

> Players who understand what's going on and take initiative to anticipate problems before they occur should be rewarded, no?

now say the same for pipe stacking :trollface:

**Comment by PJB3005** [comment]
My opinion on "you should be rewarded for placing a standalone APC" is that the "simulation precision" of SS14 isn't good enough to make that work. You can't practically have multiple independent power circuits in a room unlike real life or even games like Stationeers just due to the 2D nature of SS14 and its mechanics (eg "LV cable broadcasting effects"). 

Yeah. You can place an APC "on top" of the TEG but now you need a wall, make sure to not cross wires, deal with invisible power connections of nearby devices. None of the mechanics you are running into here feel good to play with and it's all annoying. 

I think an exception is warranted here due to the severity of the TEG shutting off on power spike.

**Comment by WarMechanic** [comment]
> Yeah. You can place an APC "on top" of the TEG but now you need a wall, make sure to not cross wires, deal with invisible power connections of nearby devices. None of the mechanics you are running into here feel good to play with and it's all annoying.

you can place an APC on top of the TEG without a wall

**Comment by WarMechanic** [comment]
> Players who understand what's going on and take initiative to anticipate problems before they occur should be rewarded, no?
yes, except the resource / time cost of placing an apc on top of the teg isn't high at all and sufficient for a single person to do such that it adds an esoteric item to an engineer's to-do list

there are a gazillion other problems an engineer can anticipate and mitigate that have a greater cost than an apc + cabling. if we're talking about power the notable things to do would be adding more connections to the hv grid throughout the station, or doubling up on substations in key departments like security or medbay which both require human interaction as people ask 'what you are doing'.

**Comment by Partmedia** [comment]
@PJB3005 approved so I'm okay with this.

File: Content.Server/Power/Generation/Teg/TegSystem.cs [hunk 1]
```diff
@@ -181,7 +181,12 @@ private void GeneratorUpdate(EntityUid uid, TegGeneratorComponent component, ref
         // Turn energy (at atmos tick rate) into wattage.
         var power = electricalEnergy * _atmosphere.AtmosTickRate;
         // Add ramp factor. This magics slight power into existence, but allows us to ramp up.
-        supplier.MaxSupply = power * component.RampFactor;
+        power *= component.RampFactor;
+
+        // Direct 1000w to power the TEG, and everything else to the station
```
Comment (Partmedia) [inline Content.Server/Power/Generation/Teg/TegSystem.cs:186]:
```suggestion
        // Simulate TEG powering itself after being started up. This means that if LV is lost this keeps running.
```

**Comment by Partmedia** [comment]
Thanks!

File: Content.Server/Power/Generation/Teg/TegSystem.cs [hunk 2]
```diff
@@ -181,7 +181,12 @@ private void GeneratorUpdate(EntityUid uid, TegGeneratorComponent component, ref
         // Turn energy (at atmos tick rate) into wattage.
         var power = electricalEnergy * _atmosphere.AtmosTickRate;
         // Add ramp factor. This magics slight power into existence, but allows us to ramp up.
-        supplier.MaxSupply = power * component.RampFactor;
+        power *= component.RampFactor;
+
+        // Simulate TEG powering itself after being started up. This means that if LV is lost this keeps running.
+        const float load = 1000;
```
Comment (PJB3005) [inline Content.Server/Power/Generation/Teg/TegSystem.cs:187]: This should not hard code the power value of the TEG, it should be based on the prototype. Now if somebody changes the value in the YML it won't work.

Comment (WarMechanic) [reply Content.Server/Power/Generation/Teg/TegSystem.cs:187]: are you going to fix this or should i

**Pull request merged**